File: presentation/src/main/java/com/example/presentation/di/AppComponent.java
package com.example.presentation.di;

import com.example.presentation.ui.fragment.MainFragment;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class,
        NetworkModule.class,
        InteractorModule.class})
public interface AppComponent {

    void inject(MainFragment injector);
}

File: presentation/src/main/java/com/example/presentation/di/AppComponent.kt
package com.example.presentation.di

import com.example.presentation.ui.fragment.MainFragment
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class, NetworkModule::class, InteractorModule::class])
interface AppComponent {
    fun inject(injector: MainFragment)
}